### ISSUE TITLE
Removing not usefull library

### ISSUE THREAD

**Issue by eldy** [Dolibarr/dolibarr#3962]
I am fighting against inclusion of external libraries that does not provide real thing.
It create dependencies and are time consuming when we check compatibilities test or feature tests when we upgrade them. Sometime, they are just deprecated, so we should n include libraries if it is not need to have dolibarr working.
This is the case for fire-php library. 200 ko added for a feature user will never see. 
You will say "so why adding phpchrome". 
Goal is to have in standard dolibarr a solution to provide support for users that are online, ask help, and have no access to their log. I could add phpchrome firephp. I choosed phpchrome because lib was just a very small file versus 200ko for fire-php. And for a "last chance" feature to provide support, having 1 solution is enough, even if it is not the fist choice of integrator. Dolibarr is en ERP not a developer toolkit.

I think if a developper or integrator need this feature, he can install its library itself. It is same for raven.
It is a good think to know we a dolibarr compatible with such a feature but i don't think we shoul have it by default into code. Risk is to see developers of module to use the embeded library when this library is not required by dolibarr so may be removed surely at the first compatibility problem (and this occurs frequently). Best solution is just to not have them into core and why not having them as an external module "Tools for integrator or support services".

I notice @rdoursenaud of this issue.

I will so probably close this issue by removing the folowing library soon:
firephp, raven, raven-js

**Comment by rdoursenaud** [comment]
The title of your issue is misleading. These are not useful **to you**.
So are a bunch of other things **to me**.

Somewhat 32Mb of TCPDF, 19Mb of CKEditor, 9.1Mb of PHPExcel and 7.2Mb of jQuery is OK but 1Mb of FirePHP, raven and raven-js is not? Come on, give me a break!

Managing libraries? That's exactly what composer is for… but you keep fighting against (learning?) it.
External libraries should not be part of the repository in the first place, but we compromised.
Build time selection could also be made to help creating tailored packages.
I won't take responsibility for bringing firephp. I just made existing code working out-of-the-box but I'll take the heat for raven and raven-js as I'm very pleased with Sentry.

Anyway, I don't want to rant forever so here's a solution:
- Just move them from the "require" section to the "suggest" section of the `composer.json`.
- Add run time presence detection of said libraries
- Problem solved :)

As for the modules, they should bring their own `composer.json` to provide said libraries. That's the whole concept. And thanks to the autoloader they won't be duplicated if versions match… Oh wait, scratch that, you don't want it and I said I'd stop ranting.
